### PR TITLE
feat: support advanced zap configuration

### PR DESCRIPTION
**Pull request by maxime1907**
*Issue number of the reported bug or feature request: #123 *

**Describe your changes**
1. Removed the nasty `DEBUG` environment variable that is loaded outside of swagger's parser.
2. Added a command line argument `--zap-config` that points to a valid json zap config file
3. Added a `/config` folder to store the zap config file
4. Added a default zap config file under `config/zap.json`

**Testing performed**
Loaded different zap configuration files such as `zap.prod.json`, `zap.dev.json` to test that the log level and outputPaths are the correct ones.

### CODE REVIEW

**Comment by maxime1907** [comment]
cc @seeker89

**Comment by seeker89** [comment]
Hey @maxime1907 thanks for the PR!

Why was the `DEBUG` gate "nasty"?

I like adding the support for zap config file, but I'd prefer goldpinger to start (say with dev defaults) if none is provided - from the quick look I had it looks like it will now fail if there is none?

**Comment by maxime1907** [comment]
> Hey @maxime1907 thanks for the PR!
> 
> Why was the `DEBUG` gate "nasty"?
> 
> I like adding the support for zap config file, but I'd prefer goldpinger to start (say with dev defaults) if none is provided - from the quick look I had it looks like it will now fail if there is none?

Hey!
I used the word nasty because i saw this comment in the code:
```
// We haven't parsed flags at this stage and that might be error prone
```

Yes it will be failing if no option is provided but i can make it work so that it loads default dev values no problem!

**Comment by maxime1907** [comment]
done! :slightly_smiling_face: @seeker89

**Comment by maxime1907** [comment]
Any updates @seeker89 ? ^^

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name ?= goldpinger
-version ?= v3.4.0
+version ?= v3.5.0
```
Comment (seeker89) [inline Makefile:2]: This will go out as `v3.7.0`

**Comment by seeker89** [comment]
Hey @maxime1907 sorry for the delay. Looks good. Please bump the version to 3.7.0 and let's get that money!

**Comment by maxime1907** [comment]
> Hey @maxime1907 sorry for the delay. Looks good. Please bump the version to 3.7.0 and let's get that money!

Done @seeker89 !

**Pull request merged**